torchgeo/datasets/landcoverai.py

```diff
@@ -170,47 +370,21 @@ def _load_target(self, id_: str) -> Tensor:
             tensor = torch.from_numpy(array).long()
             return tensor
 
-    def _verify(self) -> None:
-        """Verify the integrity of the dataset.
-
-        Raises:
-            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
-        """
-        # Check if the extracted files already exist
-        jpg = os.path.join(self.root, "output", "*_*.jpg")
-        png = os.path.join(self.root, "output", "*_*_m.png")
-        if glob.glob(jpg) and glob.glob(png):
-            return
-
-        # Check if the zip file has already been downloaded
-        pathname = os.path.join(self.root, self.filename)
-        if os.path.exists(pathname):
-            self._extract()
-            return
-
-        # Check if the user requested to download the dataset
-        if not self.download:
-            raise RuntimeError(
-                f"Dataset not found in `root={self.root}` and `download=False`, "
-                "either specify a different `root` directory or use `download=True` "
-                "to automatically download the dataset."
-            )
-
-        # Download the dataset
-        self._download()
-        self._extract()
-
-    def _download(self) -> None:
-        """Download the dataset."""
-        download_url(self.url, self.root, md5=self.md5 if self.checksum else None)
+    def _verify_data(self) -> bool:
+        """Verify if the images and masks are present."""
+        images = os.path.join(self.root, "output", "*_*.jpg")
+        masks = os.path.join(self.root, "output", "*_*_m.png")
+        return len(glob.glob(images)) > 0 and len(glob.glob(images)) == len(
```

Comment:
This globs the images list twice, let's glob it once and save that as the `images` variable to speed this up